packages-exp/auth-exp/src/core/user/reload.ts

```diff
@@ -30,23 +31,32 @@ import { getModularInstance } from '@firebase/util';
 
 export async function _reloadWithoutSaving(user: UserInternal): Promise<void> {
   const auth = user.auth;
-  const idToken = await user.getIdToken();
-  const response = await _logoutIfInvalidated(
-    user,
-    getAccountInfo(auth, { idToken })
-  );
+  let coreAccount: APIUserInfo = {};
+
+  if (user.stsTokenManager.isPassthroughMode) {
+    // The uid === localId === id token's `sub` claim
+    coreAccount.localId = user.uid;
+  } else {
+    const idToken = await user.getIdToken();
+    const response = await _logoutIfInvalidated(
+      user,
+      getAccountInfo(auth, { idToken })
+    );
 
-  _assert(response?.users.length, auth, AuthErrorCode.INTERNAL_ERROR);
+    _assert(response?.users.length, auth, AuthErrorCode.INTERNAL_ERROR);
 
-  const coreAccount = response.users[0];
+    coreAccount = response.users[0];
+  }
 
   user._notifyReloadListener(coreAccount);
 
   const newProviderData = coreAccount.providerUserInfo?.length
     ? extractProviderData(coreAccount.providerUserInfo)
     : [];
 
-  const providerData = mergeProviderData(user.providerData, newProviderData);
+  const providerData = user.stsTokenManager.isPassthroughMode
+    ? []
```

Comment:
No `custom` provider?

Reply:
That I understand, the providerData is something that we would get from `getAccountInfo`, which we're trying to circumvent in passthrough mode. I think you might be referring to the `CustomTokenProvider` (correct me if I'm wrong!) but I think the provider data and the custom token provider are only similar by name (overloading on "provider"). The `CustomTokenProvider` will be added in a subsequent PR!